test/functional/tools/remove_value.xml

```diff
@@ -0,0 +1,31 @@
+<tool id="add_remove_value" name="Add and remove values" version="0.1.0">
+    <command detect_errors="exit_code"><![CDATA[
+echo $select_value > '$out_file'
+    ]]></command>
+    <inputs>
+        <param name="select_value" type="select">
+            <options from_data_table="test_fasta_indexes">
+                <!-- contains hg18 and hg19 -->
+                <filter type="remove_value" value="hg18"/>
+                <!-- This doesn't work as we don't have the remaining values like name and path
+                <filter type="add_value" value="hg20"/>
+                -->
+            </options>
+        </param>
+    </inputs>
+    <outputs>
+        <data name="out_file"/>
+    </outputs>
+    <tests>
+        <test>
+            <output name="out_file">
+                <assert_contents>
+                    <has_line line="hg19" />
+                </assert_contents>
+            </output>
+        </test>
+        <test expect_failure="true">
```

Comment:
So this test actually passes (instead of producing a failure!) and produces an output file with `hg19`. It's not a huge  deal as that can't happen in the UI, but we should probably fix this anyway.

Reply:
I think this happens because only validators (not filters) are checked at tool execution, so it's kind of expected. Maybe comment this test out and open a separate issue, if you think that's not the correct behaviour?

Reply:
Arg, that's probably just me not knowing how to write a galaxy tool 😠. Missing the name attribute.

Reply:
yep